Automattic/wp-calypso
# HelpCenter: set env based on proxied connection

## Proposed Changes

~This uses a new API endpoint to determine if the user is proxied or not. If proxied then Happy Chat auth and chat init will connect to the staging. If not then it will use production.~

Currently there can be a discrepancy between the chat environment that is checked for availability and the chat environment that the user enters. In theory this is something only seen by proxied users, but this PR unifies the environments and allow it to change dynamically if you are proxied. We extended the `/happychat/session` endpoint to return the environment URL. This is passed through the `auth` and is used to update the `window.configData` in the happychat app.

There was also a bug introduced in a past PR that was disabling the init connection of happychat inside widgets.wp.com. I removed the bug but we need to address disabling HappyChat/inline-help in places that Help Center is showing.

## Testing Instructions

1. Open your sandbox and pull this diff - D88273-code
2. Pull branch and `cd apps/editing-toolkit/ && yarn dev --sync`
3. Sandbox a testing site URL, public-api, and widgets.wp.com
4. Go to the editor and open Help Center and start a chat.
5. Look at the websockets and ensure the check and connection use the same URL.
7. It should check against staging and connect you to staging chat.

Closes #67933
